docs/readthedocs/source/doc/Friesian/QuickStart/Train_wnd.md

```diff
@@ -0,0 +1,269 @@
+# +----+------+---+-----+----------+
+# |1   |1     |4  |345  |9         |
+# |2   |2     |3  |2775 |21        |
+# |3   |2     |2  |1971 |19        |
+# +----+------+---+-------+--------+
+```
+
+Add negative samples into the data, Friesian FeatureTable randomly choose one item as negative sample for each positive record when `neg_num=1`, `neg_num` should be larger than 1 if more negative records are needed.
+```python
+item_size = item_stats.select("item").distinct().size()
+ratings_tbl = ratings_tbl.add_negative_samples(item_size=item_size, item_col="item", label_col="label", neg_num=1)
+ratings_tbl.show(3, False)
+#
+# +----+----+----+-----+
+# |user|rate|item|label|
+# +----+----+----+-----+
+# |1   |5   |283 |0    |
+# |1   |5   |1193|1    |
+# |1   |3   |785 |0    |
+# +----+----+----+-----+
+```
+
+Add more categorical features by crossing `["gender", "age"]` into `"gender_age"`, `["age", "zipcode"]` into `"age_zipcode"` to generate more features for wide model to memorize.
+```python
+user_tbl = user_tbl.cross_columns([["gender", "age"], ["age", "zip"]], [50, 200])
+user_tbl.show(3, False)
+# 
+# +----+------+---+----+----------+----------+-------+
+# |user|gender|age|zip |occupation|gender_age|age_zip|
+# +----+------+---+-------+----------+----------+-------+
+# |1   |1     |4  |345 |9         |36        |113    |
+# |2   |2     |3  |2775|21        |46        |159    |
+# |3   |2     |2  |1971|19        |45        |135    |
+# +----+------+---+-------+----------+----------+-------+
+```
+FeatureTable can cross multiple categorical columns hash into a number of buckets by calling `feature_tbl.cross_hash_encode(columns, bins)`
+
+Join all features together and split into train and test.
+```python
+user_tbl = user_tbl.join(user_stats, on="user")
+full = ratings_tbl.join(user_tbl, on="user").join(item_stats, on="item")
+
+train_tbl, test_tbl = full.random_split([0.8, 0.2])
+full.show(3, False)
+#
+# +----+----+----+-----+------+---+----+----------+----------+-------+-----------+--------------+--------------+-----------+
+# |item|user|rate|label|gender|age|zip |occupation|gender_age|age_zip|user_visits|user_mean_rate|item_mean_rate|item_visits|
+# +----+----+----+-----+------+---+----+----------+----------+-------+-----------+--------------+--------------+-----------+
+# |148 |1088|4   |0    |1     |4  |1139|9         |36        |33     |0.5039233  |0.58825946    |0.4456522     |0.006419609|
+# |148 |5156|3   |0    |2     |6  |2261|18        |49        |172    |0.13077594 |0.81147605    |0.4456522     |0.006419609|
+# |148 |897 |5   |0    |2     |2  |1257|4         |45        |152    |0.07977332 |0.64375305    |0.4456522     |0.006419609|
+# +----+----+----+-----+------+---+-------+----------+----------+-----------+-----------+--------------+------+--------------+-----------+
+```
+
+### **2.4. Define W&D model**
+Define a W&D model using tensorflow APIs, specify feature information  using `ColumnFeatureInfo`
+```python
```

Comment:
no need to show the full details here; the focus of this tutorial is FeatureTable